kotlin/feature-layer-geodatabase/src/main/java/com/esri/arcgisruntime/sample/featurelayergeodatabase/MainActivity.kt

```diff
@@ -0,0 +1,140 @@
+/* Copyright 2020 Esri
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package com.esri.arcgisruntime.sample.featurelayergeodatabase
+
+import android.Manifest
+import android.content.pm.PackageManager
+import android.os.Bundle
+import android.util.Log
+import android.widget.Toast
+import androidx.appcompat.app.AppCompatActivity
+import androidx.core.app.ActivityCompat
+import androidx.core.content.ContextCompat
+import com.esri.arcgisruntime.data.Geodatabase
+import com.esri.arcgisruntime.layers.FeatureLayer
+import com.esri.arcgisruntime.loadable.LoadStatus
+import com.esri.arcgisruntime.mapping.ArcGISMap
+import com.esri.arcgisruntime.mapping.Basemap
+import com.esri.arcgisruntime.mapping.Viewpoint
+import kotlinx.android.synthetic.main.activity_main.*
+
+class MainActivity : AppCompatActivity() {
+  private val TAG =
+    MainActivity::class.java.simpleName
+
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+    setContentView(R.layout.activity_main)
+    // create map and add to map view
+    mapView.map = ArcGISMap(Basemap.createStreets())
+    // For API level 23+ request permission at runtime
+    if (getPermissionStatus()) loadGeodatabase()
+  }
+
+  /**
+   * Load a local geodatabase file and add it to the map
+   */
+  private fun loadGeodatabase() { // create path to local geodatabase
+    val path: String =
+      (getExternalFilesDir(null)?.path + getString(R.string.geodb_name))
+    // create a new geodatabase from local path
+    val geodatabase = Geodatabase(path)
+    // load the geodatabase
+    geodatabase.loadAsync()
```

Comment:
we can drop the loading of the geodatabase here: because you load a feature table from the geodatabase later, that then handles the load so we don't need to call it twice.  The loading and listener can work either way in that you can load and listen after, or listen before you load. So here it's ok to add a done loading listener on to the geodatabase, and we load the geodatabase via the call `geodatabaseFeatureTable.loadAsync`.

Reply:
I'm confused, when I remove `geodatabase.loadAsync()` it does not load the geodatabase and I assume that's because the feature table is only accessible from the loaded geodatabase?

Reply:
oh this is so my bad! I was comparing it to the Java SE sample and thought I'd come across loading patterns like these where you can simplify the logic but turns out this is not at all one of those, sorry for the confusion...just ignore me and leave the loadAsync where it is.

Reply:
Okay no worries!